infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -292,7 +287,9 @@ public synchronized void handleMessageLayoutPropose(CorfuPayloadMsg<LayoutPropos
 
         long serverEpoch = getServerEpoch();
 
-        if (msg.getPayload().getEpoch() != serverEpoch) {
+        // If both epoch numbers in the msg aren't equal, then the request is malformed.
+        if (msg.getPayload().getEpoch() != msg.getPayload().getLayout().getEpoch() ||
```

Comment:
Internal execution error on the server side?  Or turn this case into an ExecutionException (or other exception) that manifests on the client side?